lms/static/scripts/frontend_apps/utils/youtube.ts

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'preact/hooks';
+
+import type { ThumbnailData } from '../components/URLFormWithPreview';
+import { InvalidArgumentError } from '../errors';
+
+/**
+ * Tries to resolve the video ID from a YouTube URL
+ * @return Undefined if the ID could not be resolved
+ */
+export function videoIdFromYouTubeUrl(youTubeUrl: string): string | undefined {
+  // This regexp tries to match the video ID in the next possible URLs
+  //  * youtu.be/{id}
+  //  * {domain}/watch?v={id}[...]
+  //  * {domain}/embed/{id}[...]
+  //  * {domain}/shorts/{id}[...]
+  //  * {domain}/live/{id}[...]
+  // See https://stackoverflow.com/a/9102270 for details
+  const match = youTubeUrl.match(
+    /^.*(youtu\.be\/|v\/|u\/\w\/|embed\/|shorts\/|live\/|watch\?v=|&v=)([^#&?]*).*/
+  );
+
+  return match?.[2];
+}
+
+/**
+ * Tries to match provided URL against known YouTube video URL formats,
+ * throwing an error in case of invalid YouTube URL
+ */
+export function validateYouTubeVideoUrl(youTubeUrl: string): string {
```

Comment:
```suggestion
export function validateYouTubeVideoURL(youTubeUrl: string): string {
```

I would have renamed the param to just `url`. The function has an internal variable called `url`. I often use `parsedURL` for the `URL` constructed from a string.